raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/IpIdentityValidator.java

```diff
@@ -0,0 +1,16 @@
+package org.hypertrace.core.rawspansgrouper;
+
+import java.util.Objects;
+import org.hypertrace.core.spannormalizer.IpIdentity;
+
+public class IpIdentityValidator {
+  public static boolean isValid(IpIdentity ipIdentity) {
+    if (Objects.isNull(ipIdentity.getEnvironment()) || Objects.isNull(ipIdentity.getTenantId())) {
+      return false;
+    }
+
+    return Objects.nonNull(ipIdentity.getHostAddr())
```

Comment:
you need atleast one of host add or peer address and port for sure in my opinion